app/javascript/controllers/search_controller.js

```diff
@@ -0,0 +1,12 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+    timeout;
+    submitWithTimeout(e) {
+        const form = e.target.parentElement;
+        clearTimeout(this.timeout);
+        this.timeout = setTimeout(() => {
+            form.requestSubmit();
+        }, 400)
```

Comment:
The delay makes sense but is probably a bit too long:
According to this: https://stackoverflow.com/questions/4098678/average-inter-keypress-time-when-typing
A user notices a delay starting from 100ms.
Furthermore, the average time between keystrokes is 167ms.
I tried it with 100ms and it just feels the best. 200ms is also fine but the delay is noticeable. 
So let's do 100ms for now and if we ever encounter performance errors we know which number to bump.